marklogic-data-hub-central/ui/src/components/query-export/query-export-modal/query-export-modal.test.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { fireEvent, render } from "@testing-library/react";
+import QueryExportModal from './query-export-modal';
+
+
+describe('Query Export Component', () => {
+
+  let columns = ['id', 'firstName', 'lastName', 'age']
+
+  test('Verify Query Export Modal Dialog renders', () => {
+    const { getByTestId, getByText } = render(<QueryExportModal exportModalVisibility={true} columns={columns} />);
+    expect(getByTestId('query-export-form')).toBeInTheDocument();
+    expect(getByText('Rows:')).toBeInTheDocument();
+    expect(getByText('All')).toBeInTheDocument();
+    expect(getByText('Limited set of the first rows returned')).toBeInTheDocument();
+    expect(getByText('Export to a CSV file containing the columns of data currently displayed.')).toBeInTheDocument();
+  });
+
+  test('Verify able to select All rows', () => {
+    const { getByText } = render(<QueryExportModal exportModalVisibility={true} columns={columns} />);
+    fireEvent.click(getByText('All'));
```

Comment:
Is this expected to be selected by default? Assertion in line 35 doesnt seem to have a click event. If it is supposed to be selected by default, then this one line test again adds no value. There are no assertions here so it will always pass the click event and fail only if "All" is missing from the dom. I will delete this test.

Reply:
Thanks Srikanth, I added fix in the pr 3937.